fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraSession.java

```diff
@@ -36,9 +38,26 @@
     void expire();
 
     /**
+     * Save the session
+     *
+     * Note: use of the save method in the context of a batch operation may not persist
+     * changes to the repository until the commit method has been issued.
+     */
+    void save();
+
+    /**
      * Commit any batch operations
      */
-    void commit();
+    default void commit() {
```

Comment:
> Either way, using a default method is wrong here.

It _is_ weird/wrong, but you are being lead to do it because the type system is wrong. We have to get that right and the impl details will fit together properly, easily.